data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/buffer/KafkaBufferConfig.java

```diff
@@ -148,4 +152,12 @@ public Duration getDrainTimeout() {
     public Optional<String> getCustomMetricPrefix() {
         return Optional.ofNullable(customMetricPrefix);
     }
+
+    public CompressionConfig getCompressionConfig() {
+        if (Objects.isNull(compressionConfig)) {
```

Comment:
I don't think we want to change default behavior. We should only enable compression when compression config is explicitly provided in the YAML